client/src/components/SubTaskList/SubTaskList.js

```diff
@@ -0,0 +1,115 @@
+                <Fade in={subtasksIndex !== -1} timeout={{exit: 50}}>
+                    <Paper className={"main-paper subtasks-paper"} classes={{ root: "main-paper-root" }} elevation={0}>
+                        <Grid container alignItems="center">
+                            <Grid item xs={11}>
+                                <Typography variant="h6">{parentTask ? parentTask.content : ""}</Typography>
+                            </Grid>
+                            <Grid item xs={1}>
+                                <IconButton className={"close-button"} onClick={closeClicked}>
+                                    <CancelIcon />
+                                </IconButton>
+                            </Grid>
+                        </Grid>
+                        <Droppable droppableId="sub-task-list">
+                            {(provided) => {
+                                return (
+                                    <Paper ref={provided.innerRef} {...provided.droppableProps} className={"content-div"} id="task-list-paper" classes={{ root: "content-paper" }} elevation={0}>
+                                        <Grid container>
+                                            {subtasks.map((task, index) => {
+                                                return (
+                                                    <Draggable key={index} draggableId={`${index}`} index={index}>
+                                                        {(provided) => {
+                                                            return <SubTask
+                                                                index={index}
+                                                                task={task}
+                                                                provided={provided}
+                                                                dragging={dragging}
+                                                            />
+                                                        }}
+                                                    </Draggable>
+                                                );
+                                            })}
+                                            {provided.placeholder}
+                                        </Grid>
+                                    </Paper>
+                                )
+                            }}
+                        </Droppable>
+                        <div className={"add-button"}>
+                            <Button variant="contained" disableElevation={false} classes={{ root: "add-button-root" }} onClick={addNewTask}>Add subtask</Button>
+                        </div>
+                    </Paper>
+                </Fade>
+                <Fade in={subtasksIndex !== -1}>
+                    <div className={"subtask-div-z-0"}>
+                    </div>
+                </Fade>
+                <Fade in={subtasksIndex !== -1} timeout={{ exit: 0 }}>
```

Comment:
what's the difference between this and the one above?

Reply:
Use 2 layers of overlay when subtasks are shown. One darker overlays lies on top of all elements except subtask list and task that is currently showing subtask.

Another lighter layer overlays only the task that is currently showing subtask. So subtask list is on top of all the overlays

Reply:
Just curious, what's the rationale of having two overlays? To prevent users from editing task when subtasks are visible?

Reply:
yeah